src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,115 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_INITIAL_CAPACITY = 16;
+    private static final int DOUBLE_MULTIPLIER = 2;
+    private static final float DEFAULT_LOAD_FACTOR = 0.75f;
+    private Node<K,V>[] table;
+    private int size;
+
+    public MyHashMap() {
+        table = new Node[DEFAULT_INITIAL_CAPACITY];
+    }
 
     @Override
     public void put(K key, V value) {
-
+        int index = hash(key);
+        if (size > 0 && replaceDuplicateKeyValue(key, value)) {
+            return;
+        }
+        table = (size == table.length * DEFAULT_LOAD_FACTOR) ? resize() : table;
+        Node<K, V> current = table[index];
+        Node<K, V> newNode = new Node<>(key, value, null);;
+        if (current != null) {
+            while (current.next != null) {
+                current = current.next;
+            }
+            current.next = newNode;
+        } else {
+            table[index] = newNode;
+        }
+        size++;
     }
 
     @Override
     public V getValue(K key) {
-        return null;
+        if (!checkElementIndexExistence(key)) {
+            return null;
+        }
+        int index = hash(key);
+        Node<K, V> current = table[index];
+        while (current != null) {
+            if ((current.key == key)
+                    || (current.key != null && current.key.equals(key))) {
+                break;
+            }
+            current = current.next;
+        }
+        return current.value;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private int hash(K key) {
+        return (key == null) ? 0 : Math.abs(key.hashCode()) % table.length;
+    }
+
+    private Node<K,V>[] resize() {
+        Node<K, V>[] oldTable = table;
+        table = new Node[table.length * DOUBLE_MULTIPLIER];
```

Comment:
it's new Table so rename it

Reply:
It is not new table. It is the same table, but reseted. If you check logic of this method, you`ll see that I saved all data from my table to oldTable and reseted my current table with new length. After that I begin to adding all data with new indexes. 
I cannot create new table, because logic of my put method, that I use while adding data, is made by Node<K, V>[] table; and even if I add another some new table, program will not work properly, because method put will add data in main table.